services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -1942,6 +1944,30 @@ public void createHelixResourceAndStartMonitoring(String clusterName, String sto
     }
   }
 
+  private void createChangeCaptureVersionTopicIfChangeCaptureEnabled(
```

Comment:
So our aim in this is to have several different kinds of views which might require different kinds of resources (helix resources, kafka topics, etc.).  With the way this code is currently set up, we'd have to write this logic for every kind of view type.  I wonder if we can create a kind of factory object pattern here instead?  So basically, from a list of views in a store config, we generate a list of concrete factory instantiations which then understand how to create and manage per version resources like Kafka topics.

What do you think?